test/modules/TestResultCoordinator/Reports/LegacyTwin/LegacyTwinReportGenerator.cs

```diff
@@ -77,5 +75,78 @@ public async Task<ITestResultReport> CreateReportAsync()
             Logger.LogInformation($"Successfully finished creating LegacyTwinReport for Source [{this.SenderSource}]");
             return report;
         }
+
+        // Tolerances are needed due to a combination of false-positive failures and real product-issues.
+        // - [Nested-Edge] [Broker-Enabled]: Sometimes we get an excessive amount of 501 and 504 status codes.
+        //   Fail the tests if we have > 0.5% of either code.
+        //     + (501) We don't receive some desired properties in module-registered twin desired property callback
+        //     + (504) Module cannot make reported property update
+        bool IsPassed(IDictionary<int, int> statusCodesToCount)
+        {
+            bool isPassed = true;
+            int totalResults = statusCodesToCount.Sum(x => x.Value);
+
+            if (totalResults == 0)
+            {
+                return false;
+            }
+
+            if (this.Topology == Topology.Nested && this.MqttBrokerEnabled)
+            {
+                // See TwinTester/StatusCode.cs for reference.
+                int[] bigToleranceStatusCodes = { };
```

Comment:
I removed everything requiring a big tolerance, as the case that made me initially create this hasn't reproduced. But chose to leave it for now in case we need it later.